jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/SubscriptionResource.java

```diff
@@ -632,7 +632,7 @@ public Response doOperation(final CallContext ctx)
                 final Subscription subscription = subscriptionApi.getSubscriptionForEntitlementId(newEntitlement.getId(), ctx);
                 if (subscription.getBillingEndDate() != null) {
                     final Account account = accountUserApi.getAccountById(subscription.getAccountId(), callContext);
-                    final boolean inTheFuture = isInTheFuture(subscription.getBillingEndDate(), account);
+                    final boolean inTheFuture = isInTheFuture(subscription.getBillingEndDate().toLocalDate(), account); //TODO_1375 - Converting to LocalDate for comparison, should we instead change isInTheFuture to accept DateTime? Also, not sure how to access internalCallContext here so I've used toLocalDate
                     if (inTheFuture) {
```

Comment:
@sbrossie, could you please comment on the above too? Do, you think we should use `DateTime` here? If not, if we should use `LocalDate` how should we do the conversion? I was not able to find a way to access `internalCallContext` in the `SubscriptionResoource` class.

Reply:
Keep it the way you have it (but keep the `TODO_1375` for these choices of converting one way or the other. I would say for now, we play safe (the way you have it). keep in mind, stage 2 is to introduce new apis that will take `DateTime` as an input, so this may force us to revisit things